pkg/webui/console/containers/webhook-add/webhook-add.js

```diff
@@ -0,0 +1,192 @@
+
+  state = {
+    error: undefined,
+    displayOverwriteModal: false,
+    existingId: undefined,
+  }
+
+  @bind
+  async convertTemplateToWebhook(values) {
+    const { webhookTemplate: template, appId, createApplicationApiKey } = this.props
+    const { webhook_id, ...fields } = values
+
+    const headers = Object.keys(template.headers || {}).reduce((acc, key) => {
+      const val = template.headers[key]
+      const headerValue = val.replace(/{([a-z0-9_-]+)}/i, (_, field) => values[field])
+      if (headerValue !== '') {
+        acc[key] = headerValue
+      }
+      return acc
+    }, {})
+
+    const webhook = {
+      ids: {
+        webhook_id: values.webhook_id,
+      },
+      template_ids: template.ids,
+      format: template.format,
+      headers,
+      template_fields: fields,
+      base_url: urlTemplate.parse(template.base_url).expand(fields),
+      uplink_message: pathExpand(template.uplink_message, fields),
+      join_accept: pathExpand(template.join_accept, fields),
+      downlink_ack: pathExpand(template.downlink_ack, fields),
+      downlink_nack: pathExpand(template.downlink_nack, fields),
+      downlink_sent: pathExpand(template.downlink_sent, fields),
+      downlink_failed: pathExpand(template.downlink_failed, fields),
+      downlink_queued: pathExpand(template.downlink_queued, fields),
+      downlink_queue_invalidated: pathExpand(template.downlink_queue_invalidated, fields),
+      location_solved: pathExpand(template.location_solved, fields),
+      service_data: pathExpand(template.service_data, fields),
+    }
+
+    if (template.create_downlink_api_key) {
+      const key = {
+        name: `${webhook_id} downlink API key`,
+        rights: ['RIGHT_APPLICATION_TRAFFIC_DOWN_WRITE'],
+      }
+      const { key: downlink_api_key } = await createApplicationApiKey(appId, key)
+      webhook.downlink_api_key = downlink_api_key
+    }
+
+    return webhook
+  }
+
+  @bind
+  async existCheck(webhookId) {
+    const { appId, getWebhook } = this.props
+
+    try {
+      await getWebhook(appId, webhookId, [])
+      return true
+    } catch (error) {
+      if (isNotFoundError(error)) {
+        return false
+      }
+
+      throw error
+    }
+  }
+
+  @bind
+  async handleSubmit(webhook) {
+    const { appId, createWebhook } = this.props
+
+    await createWebhook(appId, webhook)
+  }
+
+  @bind
+  async handleSubmitSuccess() {
+    const { appId, navigateToList } = this.props
+
+    navigateToList(appId)
+  }
+
+  @bind
+  async handleWebhookSubmit(values, webhook, { setSubmitting, resetForm }) {
+    await this.setState({ error: '' })
+    try {
+      const webhookId = webhook.ids.webhook_id
+      const exists = await this.existCheck(webhookId)
+      if (exists) {
+        this.setState({ displayOverwriteModal: true, existingId: webhookId })
+        await new Promise((resolve, reject) => {
+          this.modalResolve = resolve
+          this.modalReject = reject
+        })
+      }
```

Comment:
While I said to move the whole submit handler to the container in the last review round, this `existCheck` and its result handling need to be done in the form component (while passing the `existCheck` handler down to the component). With this current solution, the container is doing UI logic by keeping state about the modal visibility. For a [proper separation of concerns](https://github.com/TheThingsNetwork/lorawan-stack/blob/v3.18/DEVELOPMENT.md#react-component-types), such UI logic is handled by the form component, so that the container can concentrate on defining logic around the necessary API requests as well as selecting, preparing and passing redux state data.

Should have been clearer about that last time.